include/vga.h

```diff
@@ -957,13 +957,18 @@ struct VGA_Type {
 	int sharpness = 0;
 };
 
-/* Hercules Palette function */
-void Herc_Palette(void);
-
-/* CGA Mono Palette function */
-void Mono_CGA_Palette(void);
+// Hercules & CGA monochrome palette
+enum class MonochromePalette : uint8_t {
+	Amber      = 0,
+	Green      = 1,
+	White      = 2,
+	Paperwhite = 3
+};
+constexpr auto MaxMonochromePalette = MonochromePalette::Paperwhite;
```

Comment:
Maybe replace the Max with the Num:
```c++
constexpr NumMonochromePalettes = auto static_cast<int>(MonochromePalette::Paperwhite) + 1
```

And could be used directly in:

```c++
auto value = (enum_val(palette) + 1) % (enum_val(MaxMonochromePalette) + 1);
```

(I didn't see other uses for the `MaxMonochromePalette`, so figured it would be a useful switch)

Reply:
Yeah, I was hesitating which one is preferable. Num is better in this case and generally more useful, I agree.